libgadget/petaio.c

```diff
@@ -210,21 +209,18 @@ void petaio_read_internal(char * fname, int ic, MPI_Comm Comm) {
         endrun(1, "Overwhelmed by part: %d > %d\n", PartManager->NumPart, PartManager->MaxPart);
     }
 
-    int newSlots[6];
-
-    for(ptype = 0; ptype < 6; ptype++) {
-        /* initialize MaxSlots to zero, such that grow don't fail. */
-        newSlots[ptype] = 0;
-        if(NLocal[ptype] > 0) {
-            newSlots[ptype] = All.PartAllocFactor * NLocal[ptype];
-        }
-    }
-
     /* Now allocate memory for the secondary particle data arrays.
      * This may be dynamically resized later!*/
 
     /*Ensure all processors have initially the same number of particle slots*/
-    MPI_Allreduce(MPI_IN_PLACE, newSlots, 6, MPI_INT, MPI_MAX, Comm);
+    int newSlots[6] = {0};
```

Comment:
Could you add a comment refering to impi version and this ticket, so we don't roll this back accidentally?

Reply:
No ticket, but I added a comment with the version